src/lib/uefi/src/lib.rs

```diff
@@ -209,11 +209,10 @@ where
         }
 
         // Iterate through files.
-        let res = {
+        while {
```

Comment:
Why do you change this. Did this use to work?

Reply:
It probably use to break tests. I remember.

Reply:
Yes, it's restored from the original implementation, see the commit message.